vmm/src/vm.rs

```diff
@@ -1041,6 +1103,11 @@ impl Vm {
                 let kernel = File::open(kernel).map_err(Error::KernelFile)?;
                 Self::load_kernel(None, Some(kernel), memory_manager)
             }
+            (Some(firmware), Some(kernel)) => {
+                let firmware = File::open(firmware).map_err(Error::FirmwareFile)?;
+                let kernel = File::open(kernel).map_err(Error::KernelFile)?;
+                Self::load_kernel(Some(firmware), Some(kernel), memory_manager)
+            }
```

Comment:
Can we achieve the booting with both firmware and kernel simply in this way?
- Keep the implementation of `load_kernel`
- Invoke `Self::load_kernel(None, Some(kernel), memory_manager);` to load the kernel at first and invoke `Self::load_kernel(Some(firmware), None, memory_manager)` and return the `EntryPoint` of the firmware